Source/WebKit/NetworkProcess/Classifier/WebResourceLoadStatisticsStore.cpp

```diff
@@ -473,6 +474,26 @@ void WebResourceLoadStatisticsStore::requestStorageAccess(RegistrableDomain&& su
     });
 }
 
+void WebResourceLoadStatisticsStore::setLoginStatus(RegistrableDomain&& domain, IsLoggedIn loggedInStatus, CompletionHandler<void()>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+
+    if (loggedInStatus == IsLoggedIn::Yes)
+        m_loginStatus.set(domain, loggedInStatus);
+    else
+        m_loginStatus.remove(domain);
+    completionHandler();
+}
+
+void WebResourceLoadStatisticsStore::isLoggedIn(RegistrableDomain&& domain, CompletionHandler<void(bool)>&& completionHandler)
+{
+    ASSERT(RunLoop::isMain());
+
+    if (m_loginStatus.contains(domain))
+        return completionHandler(m_loginStatus.get(domain) == IsLoggedIn::Yes);
```

Comment:
This is doing a double HashMap lookup (once in `contains()` above and then again in `get()` here). I recommend using HashMap::find() and use the iterator to avoid the double hash look up.

Reply:
e.g.
```
auto it = m_loginStatus.find(domain);
completionHandler(it != m_loginStatus.end() && it->value == IsLoggedIn::Yes);
```